vendor/bat-native-ads/src/bat/ads/internal/experiments/experiments.cc

```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/experiments/experiments.h"
+
+#include <string>
+
+#include "bat/ads/internal/ads_impl.h"
+#include "bat/ads/internal/experiments/experiments_util.h"
+#include "bat/ads/internal/experiments/page_probabilities/page_probabilities_experiment.h"
+#include "bat/ads/internal/logging.h"
+
+namespace ads {
+
+Experiments::Experiments(
+    AdsImpl* ads)
+    : ads_(ads) {
+  DCHECK(ads_);
+}
+
+Experiments::~Experiments() = default;
+
+void Experiments::Log() {
+  experiments::PageProbabilities page_probabilities_experiment;
```

Comment:
why would the superclass have to know about a subclass? In any case I think the class should probably just go away per the other comments

Reply:
`Experiments` is not the superclass but the way we communicate between `AdsImpl` and individual experiments. Aside from that every experiment under `experiments/*` adheres to the interface defined in `experiment.h`.

Reply:
I see, it's Experiments vs Experiment. For future reference this should generally be something like ExperimentList to avoid that kind of confusion

Reply:
resolved with refactor